Firestore/core/src/nanopb/nanopb_util.h

```diff
@@ -104,6 +104,69 @@ T* _Nonnull MakeArray(pb_size_t count) {
   return static_cast<T*>(calloc(count, sizeof(T)));
 }
 
+template <typename T>
+T* _Nonnull ResizeArray(void* _Nullable ptr, pb_size_t count) {
+  return static_cast<T*>(realloc(ptr, count * sizeof(T)));
+}
+
+/**
+ * Initializes a repeated field with a list of values. Applies `converter` to
+ * each value before assigning.
+ */
+template <typename T, typename Iterator, typename Func>
+void SetRepeatedField(T** fields_array,
+                      pb_size_t* fields_count,
+                      Iterator first,
+                      Iterator last,
+                      Func converter) {
+  *fields_count = CheckedSize(std::distance(first, last));
+  *fields_array = MakeArray<T>(*fields_count);
+  auto* current = *fields_array;
+  while (first != last) {
+    *current = converter(*first);
+    ++current;
+    ++first;
+  }
+}
+
+/**
+ * Initializes a repeated field with a list of values. Applies `converter` to
+ * each value before assigning.
+ */
+template <typename T, typename Container, typename Func>
+void SetRepeatedField(T** fields_array,
+                      pb_size_t* fields_count,
+                      const Container& fields,
+                      Func converter) {
+  *fields_count = CheckedSize(fields.size());
+  *fields_array = MakeArray<T>(*fields_count);
+  auto* current = *fields_array;
+  for (const auto& field : fields) {
+    *current = converter(field);
+    ++current;
+  }
+}
+
+/** Initializes a repeated field with a list of values. */
+template <typename T>
+void SetRepeatedField(T** fields_array,
+                      pb_size_t* fields_count,
+                      const std::vector<T>& fields) {
+  return SetRepeatedField(fields_array, fields_count, fields,
+                          [](const T& val) { return val; });
+}
+
+/**
+ * Zeroes out the memory of `fields`. This can be used if the contents of fields
+ * array are moved to another message that takes on ownership.
+ */
+template <typename T>
+void ReleaseFieldsArray(T* _Nonnull fields, pb_size_t fields_count) {
```

Comment:
Nit: the name can be understood as releasing (i.e., freeing) the array itself, not its members. Also, "release" can mean "free" rather than "let go of ownership". Consider something like `DisownFieldsArrayMembers`.